btcclient/client_block_subscriber.go

```diff
@@ -56,7 +57,7 @@ func NewWithBlockSubscriber(cfg *config.BTCConfig, retrySleepTime, maxRetrySleep
 		if err != nil {
 			return nil, fmt.Errorf("failed to get BTC backend: %v", err)
 		}
-		if backend != rpcclient.BitcoindPost25 {
+		if !strings.EqualFold(backend.String(), rpcclient.BitcoindPost25.String()) {
 			return nil, fmt.Errorf("zmq is only supported by bitcoind, but got %v", backend)
 		}
```

Comment:
This check is to ensure we use bitcoind not btcd, since only bitcoind supports zmq that is necessary for the vigilante. So as long as the type matches it's okay. The version of bitcoind is not necessarily 25 or after

Reply:
So can we do sth like

```
		if !strings.Contains(backend.String(), "bitcoind") {
			return nil, fmt.Errorf("zmq is only supported by bitcoind, but got %v", backend)
		}
```

Reply:
modified, a bit different, but tested in deployments :+1: